code/scripts/DataTableGen.hs

```diff
@@ -0,0 +1,292 @@
+      ntEntries = zipWith join' ntEntryHds ntRefLines
+
+  -- creates file entry data by combining data, newtype and class entries
+  let entryData = dtEntries ++ ntEntries ++ clsEntries
+  
+  -- guards determine how to handle overall file entry output
+  -- for single line entry data vs. multi-line entry data
+  let output 
+        | length entryData > 1 && not (null dtRefLines) = fdtl ++ "\n" ++ sbE
+        | length entryData > 1 && not (null ntRefLines) = fntl ++ "\n" ++ sbE
+        | length entryData > 1 = entry ++ "\n" ++ sbE
+        | length entryData == 1 && not (null dtRefLines) = fdtl
+        | length entryData == 1 && not (null ntRefLines) = fntl
+        | otherwise = entry
+      sbE = intercalate "\n" (drop 1 entryData)
+      fdtl = join' entry (head dtRefLines)
+      fntl = join' entry (head ntRefLines)
+
+  -- mapM_ print (lines output)
+  return output
+
+-- used to filter out info lines (i.e. removes comment and empty lines)
+isInfoLine :: String -> Bool
+isInfoLine line = (line /="") && not ("#" `isPrefixOf` line)
+
+-- converts string to classtype (for use by config function)
+toClassType :: String -> ClassType
+toClassType "Haskell" = Haskell
+toClassType "Drasil" = Drasil
+toClassType "GOOL" = GOOL
+
+-- gets folder from dictionary using folder name (iff it exists in dictionary)
+getFolder :: Map.Map DC.FolderName DC.Folder -> DC.FolderName -> DC.Folder
+getFolder dict name = fromJust $ Map.lookup name dict
+
+-- converts list to dictionary list format (for use by drasil- directories only)
+toDictList :: DC.Folder -> (DC.FolderName,DC.Folder)
+toDictList folder = (DC.folderDrasilPack folder,folder) 
+
+-- gets raw Entries, extracts classes and orders them with config file settings
+ordClasses :: [ClassType] -> [Entry] -> [Class]
+ordClasses cIO rED = concatMap (getClasses hslCls drlCls glCls) cIO where
+  (hslCls,drlCls,glCls) = sortClasses dfCls instClsNms
+  -- dfCls = all defined classes; instClsNms = all instanced class names
+  dfCls = concatMap classes rED
+  instClsNms = nub . map clsInstName $ concatMap classInstances rED
+
+-- gets lists of all defined/instanced classes; sorts by class type
+sortClasses :: [Class] -> [ClassName] -> ([Class],[Class],[Class])
+sortClasses dCls iClsN = (haskellCls,drasilCls,goolCls) where
+  -- partitions defined classes into GOOL and Drasil type classes
+  (goolCls,drasilCls) = partition (isTypeOf GOOL) dCls
+  -- gets defined classes names; used to isolate for haskell classes
+  goolClsNms = map className goolCls
+  drasilClsNms = map className drasilCls
+  haskellClsNms = iClsN \\ (goolClsNms ++ drasilClsNms)
+  -- makes new Haskell type classes
+  haskellCls = map (makeClass Haskell) haskellClsNms
+
+-- outputs class instance group; used by ordClasses to order the class instances 
+-- by class type (as defined in config file settings)
+getClasses :: [Class] -> [Class] -> [Class] -> ClassType -> [Class]
+getClasses h d g Haskell = h
```

Comment:
use `_` for unused parameters (like `d` and `g` on this line).

Reply:
I've made the changes to function `getClasses` [here](https://github.com/JacquesCarette/Drasil/pull/2208/commits/4065330bf5885d36437fb97d322a5c23d339df61#) (old lines 260-262; new lines 259-261).